packages/master-detail-layout/src/vaadin-master-detail-layout.js

```diff
@@ -417,6 +421,26 @@ class MasterDetailLayout extends SlotStylesMixin(ResizeMixin(ElementMixin(Themab
     }
   }
 
+  /** @private */
+  __onBackdropClick() {
+    this.dispatchEvent(
+      new CustomEvent('backdrop-click', {
+        bubbles: true,
+      }),
+    );
+  }
+
+  /** @private */
+  __onDetailKeydown(event) {
+    if (event.key === 'Escape') {
```

Comment:
Similarly we might want to "consume" there event here, so that it doesn't cause an outer details to close.